include/mapbox/variant.hpp

```diff
@@ -645,12 +645,14 @@ class variant
         // move_assign uses move-construction via placement new.
         noexcept(detail::conjunction<std::is_nothrow_move_constructible<Types>...>::value)
     {
+        if (this == &other) return *this; // check for self-assignment
         move_assign(std::move(other));
         return *this;
     }
 
     VARIANT_INLINE variant<Types...>& operator=(variant<Types...> const& other)
     {
+        if (this == &other) return *this; // check for self-assignment
```

Comment:
nit:
```
if (this != &other) {
    move_assign(std::move(other));
}
return *this;
```

Reply:
Did you mean using move_assign(..) in copy assignment op? 
+1 ,  on style
```c++
if (this != &other) 
{
    ....
}
return *this;
```

Reply:
> Did you mean using move_assign(..) in copy assignment op?

nope, leftover, just meant the style change :smile: